.github/actions/require-multiple-reviewers-v1/src/run.ts

```diff
@@ -0,0 +1,72 @@
+import { getAnnotations, getImportantFilesChanged } from './utils'
+import type { Conclusion, Core, GitHub } from './types'
+
+export default async function run(core: Core, github: GitHub): Promise<void> {
+  try {
+    if (!github.context.payload.pull_request) {
+      throw new Error(
+        'This action can only be run in the context of a pull request.'
+      )
+    }
+
+    const numberOfReviewers = parseInt(
+      core.getInput('number-of-reviewers', { required: true }),
+      10
+    )
+    if (isNaN(numberOfReviewers)) {
+      throw new Error('number-of-reviewers input is not a number')
+    }
+
+    const token = core.getInput('token', { required: true })
+    const octokit = github.getOctokit(token)
+    const { owner, repo } = github.context.repo
+
+    const { data: files } = await octokit.rest.pulls.listFiles({
+      owner,
+      repo,
+      pull_number: github.context.payload.pull_request.number
+    })
+
+    const changedFiles = files
+      .filter(file => file.status !== 'unchanged')
+      .map(file => file.filename)
+
+    const importantFilesChanged = getImportantFilesChanged(
+      core.getInput('important-files-path', {
+        required: true
+      }),
+      changedFiles
+    )
+
+    const requiresMultipleReviewers = importantFilesChanged.length > 0
+    let conclusion: Conclusion = 'neutral'
+
+    if (requiresMultipleReviewers) {
+      const { data: reviews } = await octokit.rest.pulls.listReviews({
+        owner,
+        repo,
+        pull_number: github.context.payload.pull_request!.number
+      })
+
+      const approvals = reviews.filter(review => review.state === 'APPROVED')
+      conclusion = approvals.length < numberOfReviewers ? 'failure' : 'success'
+    }
+
+    await octokit.request('POST /repos/{owner}/{repo}/check-runs', {
```

Comment:
Did you test that this does the right thing in the event that we already left a check for the same commit? eg, if it's triggered from a `pull_request` and then later a `pull_request_review` trigger, did you verify that the second one looks like it's updating the original check status (as opposed to leaving a new second status)? What about the annotations - do they get cleared out when posting the updated check, or do they need to be removed explicitly?

(I'd recommend trying this out in a separate test repo and linking the results of your testing in the PR description)

Reply:
I'ts going to fail and re-run the `Tests / require-multiple-reviewers` action 

<img width="844" alt="image" src="https://github.com/user-attachments/assets/48ee1251-4888-4279-a572-413ea0adf4fd" />

Annotations are warnings and don't go anywhere, they are for the information purpose so a PR author knows what important files are, and I don't think they should be removed.